hazelcast/src/main/java/com/hazelcast/map/impl/MapDataSerializerHook.java

```diff
@@ -98,82 +118,100 @@
-    public static final int LOAD_ALL = 18;
-    public static final int ENTRY_BACKUP = 19;
-    public static final int ENTRY_OPERATION = 20;
-    public static final int PUT_ALL = 21;
-    public static final int PUT_ALL_BACKUP = 22;
-    public static final int REMOVE_IF_SAME = 23;
-    public static final int REPLACE = 24;
-    public static final int SIZE = 25;
-    public static final int CLEAR_BACKUP = 26;
-    public static final int CLEAR_NEAR_CACHE = 27;
-    public static final int CLEAR = 28;
-    public static final int DELETE = 29;
-    public static final int EVICT = 30;
-    public static final int EVICT_ALL = 31;
-    public static final int EVICT_ALL_BACKUP = 32;
-    public static final int GET_ALL = 33;
-    public static final int IS_EMPTY = 34;
-    public static final int MERGE = 35;
-    public static final int NEAR_CACHE_SINGLE_INVALIDATION = 36;
-    public static final int NEAR_CACHE_BATCH_INVALIDATION = 37;
-    public static final int CHECK_IF_LOADED = 38;
-    public static final int PARTITION_WIDE_ENTRY = 39;
-    public static final int PARTITION_WIDE_ENTRY_BACKUP = 40;
-    public static final int PARTITION_WIDE_PREDICATE_ENTRY = 41;
-    public static final int PARTITION_WIDE_PREDICATE_ENTRY_BACKUP = 42;
-    public static final int ADD_INDEX = 43;
-    public static final int AWAIT_MAP_FLUSH = 44;
-    public static final int CONTAINS_VALUE = 45;
-    public static final int GET_ENTRY_VIEW = 46;
-    public static final int FETCH_ENTRIES = 47;
-    public static final int FETCH_KEYS = 48;
-    public static final int FLUSH_BACKUP = 49;
-    public static final int FLUSH = 50;
-    public static final int MULTIPLE_ENTRY_BACKUP = 51;
-    public static final int MULTIPLE_ENTRY = 52;
-    public static final int MULTIPLE_ENTRY_PREDICATE_BACKUP = 53;
-    public static final int MULTIPLE_ENTRY_PREDICATE = 54;
-    public static final int NOTIFY_MAP_FLUSH = 55;
-    public static final int PUT_IF_ABSENT = 56;
-    public static final int PUT_FROM_LOAD_ALL = 57;
-    public static final int PUT_FROM_LOAD_ALL_BACKUP = 58;
-    public static final int QUERY_PARTITION = 59;
-    public static final int QUERY = 60;
-    public static final int PUT_TRANSIENT = 61;
-    public static final int REPLACE_IF_SAME = 62;
-    public static final int TRY_PUT = 63;
-    public static final int TRY_REMOVE = 64;
-    public static final int TXN_LOCK_AND_GET = 65;
-    public static final int TXN_DELETE = 66;
-    public static final int TXN_PREPARE = 67;
-    public static final int TXN_PREPARE_BACKUP = 68;
-    public static final int TXN_ROLLBACK = 69;
-    public static final int TXN_ROLLBACK_BACKUP = 70;
-    public static final int TXN_SET = 71;
-    public static final int TXN_UNLOCK = 72;
-    public static final int TXN_UNLOCK_BACKUP = 73;
+    public static final MutableInteger ID = new MutableInteger();
 
-    private static final int LEN = TXN_UNLOCK_BACKUP + 1;
+    public static final int PUT = ID.value++;
```

Comment:
This is very dangerous. I think the existing approach is more solid since you hard code a id to a type of class and therefor it is easier to see which id is mapped to which class + it more difficult to f*ck up. If someone removes e.g. ENTRIES, then all id's are now magically going to shift. With the existing approach that would not happen.